### DISCUSSION TITLE
Does it produce Api endpoints?

### DISCUSSION THREAD

**Discussion by rikousik** [serverpod/serverpod#100] [Q&A] [answered]
Hi, I understand that it generates client code that's directly usable in flutter project. But what if I want to use the same server in my website? Does it also generates Api-endpoints to do that ?

**Comment by vlidholt** [comment, marked as answer]
Serverpod does generate well-formed JSON endpoints, however the format is not yet documented so it may be a little tricky to figure out exactly what the API looks like. Serverpod also comes with a web server (serverpod_relic). It's fully working, but still a bit experimental, but that is the planned path to support web pages with Serverpod.

**Reply by iampawan** [reply]
Can you share a sample for accessing the API. Like, for ExampleEndpoint there is a hello method. How the API will look like for `hello`?

**Reply by MarkOSullivan94** [reply]
@iampawan if you are wanting to test out `ExampleEndpoint` after getting Serverpod up and running, you can execute the below cURL command.

I'm pretty sure both Postman and Insomnia supports importing cURL commands so hopefully this should let you also test the endpoint on them too.

```curl
curl --request GET \
  --url http://localhost:8080/example \
  --header 'Content-Type: application/json' \
  --data '{
	"method": "hello",
	"name": "world"
}'
```

Note: I was running Serverpod `0.9.5` for this, so if this is no longer working it may be the case that the example has changed.

**Reply by vlidholt** [reply]
Out of curiosity, what is your use case for wanting to access the API through a GET request / curl? Just to test that the server is up and running or something else?

That being said, there will be documentation added for the protocol itself.

**Reply by MarkOSullivan94** [reply]
Always useful being able to test server endpoints via Insomnia / Postman - I know I always resort to trying out APIs in Insomnia / Postman first before implementing them in an app

**Reply by Sameer330** [reply]
So I've started learning Serverpod on 22 Jan, 2024. I've been finding ways to create API endpoints that can be used in the browser or postman. Here's what I have tried:

In postman:

Method: POST
URL: https://localhost:8080/example
BODY: 
{
  "method": "hello",
  "name": "Sameer"
}

Yet, I didn't get the expected response on postman.

Can anyone help me with creating API endpoints in serverpod that work similar to JavaScript API endpoints.

**Comment by rikousik** [comment]
Okay, I found a way to access endpoints of serverpod (not relic). This is not a very clean approach but it's working.

Endpoint: client.example.hello("world");

should try like this: 

http.post( baseUrl+"/example",
body: {
method: hello,
name: world
}
);

**Comment by julianoventola** [comment]
http://localhost:8080/endpoint_name
Using post put this in the body (the method is the name of the internal endpoint's method:
{
	"method":"create",
  "name": "John doe",
 }

**Comment by ameerPillsbee** [comment]
how to access endpoints which require signin with postman?? how to include the key in the request?
